src/decode.rs

```diff
@@ -1317,7 +1318,8 @@ fn decode_b(
         return Ok(());
     }
 
-    let ts_c = &mut *ts.context.try_lock().unwrap();
+    // Available if t.frame_thread.pass != 2
+    let ts_c = &mut **ts_c.as_mut().unwrap();
```

Comment:
This can be combined with the `if t.frame_thread.pass == 2 {` that is for the block ending in the `return Ok(())`.